### PR TITLE
Cover WinVFS in testcases

### PR DESCRIPTION
**Pull request by erikjv**

### CODE REVIEW

File: test/testsyncmove.cpp [hunk 1]
```diff
@@ -841,21 +841,73 @@ private slots:
...
+        {
+            auto localState = fakeFolder.currentLocalState();
+            QVERIFY(localState.find("A/file") == nullptr); // check if the file is gone
+            QVERIFY(localState.find("A") == nullptr); // check if the directory is gone
+            FileInfo *file = localState.find({ "B/file" });
+            QVERIFY(file != nullptr); // check if the file exists
+            if (vfsMode != Vfs::Off) {
+                QVERIFY(file->isDehydratedPlaceholder); // check that no-one messed with the placeholder state
+            }
+            QVERIFY(localState.equals(fakeFolder.currentRemoteState(), cmpOpt));
```
Comment (TheOneRing) [inline test/testsyncmove.cpp:909]: The benefit of a QCOMPARE here is that we had human readable errors in the log

File: test/testutils/syncenginetestutils.cpp [hunk 2]
```diff
@@ -254,9 +255,38 @@ bool FileInfo::operator==(const FileInfo &other) const
...
+            && children == other.children;
+}
+
+bool FileInfo::equals(const FileInfo &other, ComparissonOption opt) const
+{
+    switch (opt) {
+    case ContentIsKing:
+        return *this == other;
+    case IgnoreContentOfDehydratedFiles:
+        if (!isDehydratedPlaceholder && !other.isDehydratedPlaceholder) {
```
Comment (TheOneRing) [inline test/testutils/syncenginetestutils.cpp:269]: please also check modtime and other available meta data.

File: test/testsyncmove.cpp (hunk 1 above)
Comment (TheOneRing) [reply test/testsyncmove.cpp:909]: hmm would be awesome if we could wrap all this complexity behind some magic else our tests complexity will increase exponential 🤯

Comment (erikjv) [reply test/testsyncmove.cpp:909]: I'm wondering if we should get rid of the old `==` implementation. But I don't know how many tests that'd break.

File: test/testutils/syncenginetestutils.cpp (hunk 2 above)
Comment (erikjv) [reply test/testutils/syncenginetestutils.cpp:269]: which ones, other than `lastModified`?

Comment (TheOneRing) [reply test/testutils/syncenginetestutils.cpp:269]: Well you are right we only have the modtime.

**Review by TheOneRing — APPROVED** [review]

**Pull request merged**